lib/compressTextures.js

```diff
@@ -0,0 +1,508 @@
+            cliFormat = 'RGB8A1';
+        } else if (transparent && !alphaBit) {
+            cliFormat = 'RGBA8';
+        } else if (!transparent) {
+            cliFormat = 'RGB8';
+        }
+    }
+
+    var cpuCount = os.cpus().length;
+    var cliOptions = [inputPath, '-format', cliFormat, '-effort', quality, '-jobs', cpuCount, '-output', outputPath];
+
+    return runCompressTool(etc2compPath, cliOptions, outputPath)
+        .then(function(buffer) {
+            return {
+                buffer : buffer,
+                extension : extension
+            };
+        });
+}
+
+function compressWithPVRTexTool(inputPath, tempDirectory, options) {
+    var extension = '.ktx';
+    var outputPath = getTempImagePath(tempDirectory, extension);
+
+    var format = options.format;
+    var quality = Math.floor(options.quality / 2.1); // Map quality to a 0-4 scale
+    var bitrate = options.bitrate;
+    var transparent = options.transparent;
+
+    var qualityOptions = ['pvrtcfastest', 'pvrtcfast', 'pvrtcnormal', 'pvrtchigh', 'pvrtcbest'];
+    var cliQuality = qualityOptions[quality];
+    var cliFormat;
+
+    if (format === 'pvrtc1') {
+        if (transparent && bitrate === 2) {
+            cliFormat = 'PVRTC1_2';
+        } else if (transparent && bitrate === 4) {
+            cliFormat ='PVRTC1_4';
+        } else if (!transparent && bitrate === 2) {
+            cliFormat = 'PVRTC1_2_RGB';
+        } else if (!transparent && bitrate === 4) {
+            cliFormat = 'PVRTC1_4_RGB';
+        }
+    } else if (format === 'pvrtc2') {
+        if (bitrate === 2) {
+            cliFormat = 'PVRTC2_2';
+        } else if (bitrate === 4) {
+            cliFormat = 'PVRTC2_4';
+        }
+    }
+
+    // No CPU count - this tool is single-threaded
+    var cliOptions = ['-i', inputPath, '-o', outputPath, '-f', cliFormat, '-q', cliQuality, '-square', '-', '-pot', '-'];
+
+    return runCompressTool(pvrTexToolPath, cliOptions, outputPath)
+        .then(function(buffer) {
+            return {
+                buffer : buffer,
+                extension : extension
+            };
+        });
+}
+
+function compressWithCrunch(inputPath, tempDirectory, options) {
+    var format = options.format;
+    var transparent = options.transparent;
+
+    var extension;
+    var cliFormat;
+    var fileFormat;
+
+    if (format.indexOf('crunch') >= 0) {
+        extension = '.crn';
+        fileFormat = 'crn';
+        format = format.slice(7);
+    } else {
+        extension = '.ktx';
+        fileFormat = 'ktx';
+    }
+
+    if (format === 'dxt1') {
+        if (transparent) {
+            cliFormat = '-DXT1A';
+        } else {
+            cliFormat = '-DXT1';
+        }
+    } else if (format === 'dxt3') {
+        cliFormat = '-DXT3';
+    } else if (format === 'dxt5') {
+        cliFormat = '-DXT5';
+    }
+
+    var outputPath = getTempImagePath(tempDirectory, extension);
+    var cpuCount = os.cpus().length;
```

Comment:
Add a tasklist item (that will go into the roadmap) to make this an option, which we'll want for server-side.